doc_src/index.hdr.in

```diff
@@ -17,7 +17,201 @@
+\endfish
+
+This convention helps to make sure help can be found on commands no matter where they originate from. *fish*'s internal commands all come with a manual page.
+
+\subsection shell-syntax Command Syntax
+
+Shell commands also have some syntax in common, no matter where they originate from: 
+
+Commands consist of three parts:
+
+\fish
+COMMAND [OPTIONS] ARGUMENTS
+\endfish
+
+- <b>`COMMAND`</b>: the name of the executeable
+
+- <b>`[OPTIONS]`</b>: options can change the behaviour of a command
+
+- <b>`ARGUMENTS`</b>: input to the command
+
+For external <b>commands</b> the executable is usually stored on file and the file path must be included in the variable `$PATH`, so that the file can be found: see <a href="#variables-special">Special Variables</a>.
+
+<b>Options</b> come in two forms: a short name, that is a hyphen with a single letter; or a long name, consisting of two hyphens with words connected by hyphens. Example: `-h` and `--help` are common options for opening the manual page of a command. Options are a fixed set, described in the manual pages of the command, see <a href="#man-page">Manual Pages</a>
+
+<b>Arguments</b> are the arbitrary input part of a command: often it is a file or directory name, sometimes it is a string or a list.
+
+Example:
+
+\fish
+>echo -s Hallo World!
+HalloWorld!
+\endfish
+
+-  both `Hallo` and `World!` are arguments to the echo command
+- `-s` is an option that suppresses spaces in the output of the command
+
+\subsection shell-programming Commands versus Programs
+
+Shell commands differ from other programs in the following way:
+
+- they cannot return variables
+
+Instead:
+
+- They transform an <b>input stream</b> into an <b>output stream</b>. Both of these streams are usually the terminal, but they can be redirected.
+
+- They return an <b>exit code</b>: this exit code is 0 when the command executes normally and between 1 and 255 otherwise.
+
+This leads to another way of programming for shell commands: they can directly return variables, but they can pass on content to other commands in following ways:
+
+<b>Commands can pass on content as a stream</b>: one command takes the output stream of another command as its input stream: 
+
+Example:
+
+\fish
+>ls -l | grep "my topic"
+\endfish
+
+- every line of the `ls` command is immediatelly passed on to the `grep` command and treated separately. 
+
+In this case the two commands execute in parallel. This is called piping, see <a href="#piping">Piping</a>.
+
+<b>Commands can pass on all of their output stream as a chunk</b>: 
+
+Example:
+\fish
+>echo (ls a*)
+\endfish
+
+- the `echo` command takes the output of the `ls` command as a chunk and prints it to the terminal. The `echo` command waits for the `ls` command to finish, before it executes.
+
+In this case one command waits for the other command to finish, before it executes. This is called command substitution, see <a href="#expand-command-substitution">Command Substitution</a>.
```

Comment:
How exact do we want to be here? _fish_ waits for the inner command to finish, the outer one does not exist until _fish_ executes it.

Is that distinction worth it?

Reply:
I left these details out. You are right it is not worth this distinction here.